File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
         int len = nums.length;

        int right = len - 1;
        int left = 0;

        while (left < right) {
            int mid = (left + right) / 2;

            if (nums[right] > nums[mid]) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        return nums[right];
    }
}

File: Data Structures & Algorithms/find-pivot-index/submission-0.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        
        def findSum(left, right):
            subSum = 0
            for i in range(left, right):
                subSum += nums[i]
            
            return subSum
        
        for i in range(len(nums)):
            lSum = findSum(0, i)
            rSum = findSum(i+1, len(nums))

            if lSum == rSum:
                return i
        
        return -1